hivemind/server/connection_handler.py

```diff
@@ -58,13 +58,18 @@ async def info(self, request: runtime_pb2.ExpertUID, context: grpc.ServicerConte
         return runtime_pb2.ExpertInfo(serialized_info=pickle.dumps(self.experts[request.uid].get_info()))
 
     async def forward(self, request: runtime_pb2.ExpertRequest, context: grpc.ServicerContext):
+        # TODO: add loading compression parameter
         inputs = [deserialize_torch_tensor(tensor) for tensor in request.tensors]
         future = self.experts[request.uid].forward_pool.submit_task(*inputs)
-        serialized_response = [serialize_torch_tensor(tensor) for tensor in await future]
+        forward_schema_flatten = list(nested_flatten(self.experts[request.uid].forward_schema))
+        serialized_response = [serialize_torch_tensor(tensor, forward_schema_flatten[i].compression)
+                               for i, tensor in enumerate(await future)]
```

Comment:
```suggestion
        forward_schema_flatten = list(nested_flatten(self.experts[request.uid].forward_schema))
        serialized_response = [serialize_torch_tensor(tensor, proto.compression) tensor, proto in zip(
                               await future, nested_flatten(self.experts[request.uid].forward_schema))]
```

Can we re-write this with iterators instead of instantiating a list?